shell.nix

```diff
@@ -0,0 +1,74 @@
+{ type ? "develop" }:
+let
+  nixpkgs = let cfg = builtins.fromJSON (builtins.readFile ./sources.json);
+  in fetchTarball {
+    url = "https://github.com/${cfg.owner}/${cfg.repo}/tarball/${cfg.rev}";
+    sha256 = cfg.sha256;
+  };
+  pkgs = import nixpkgs {
+    config = { allowUnfree = true; };
+    overlays = [ ];
+  };
+  inherit (pkgs) lib;
+
+  # TODO: Remove this when upgrading to nixos-20.03 when we can
+  # use poetry from the stable channel
+  unstableSrc = pkgs.srcOnly {
+    name = "nixpkgs-unstable-src";
+    src = pkgs.fetchzip {
+      url = "https://github.com/nixos/nixpkgs/tarball/76e1d5cf3b8f817e4e43f5b0fcdb951426956c35";
+      sha256 = "27f3c5a771733c68f1e84c18575fa11e206167564685756d9000ef7f9e15e311";
+    };
+  };
+  unstable = import unstableSrc {
```

Comment:
No need to do the `srcOnly` thing if it doesn't need any patches. Can use `unstable = import (fetchTarball { url = "..."; sha256 = "..."; }) { ... }` instead. Note that the `sha256` might be different between `fetchTarball` and `fetchzip`.

Reply:
Got
```
unpacking 'https://github.com/NixOS/nixpkgs/tarball/1dadefec574531f021bddc85ca3c9bc81fd52899'...
unpacking 'https://github.com/nixos/nixpkgs/tarball/76e1d5cf3b8f817e4e43f5b0fcdb951426956c35'...
error: while evaluating the attribute 'buildInputs' of the derivation 'dev-shell' at /home/builder/conduit/shell.nix:57:3:
while evaluating 'getOutput' at /nix/store/dvaz934b0ya11aq9x2ma1fc81fa3jpjr-source/lib/attrsets.nix:464:23, called from undefined position:
while evaluating anonymous function at /nix/store/dvaz934b0ya11aq9x2ma1fc81fa3jpjr-source/pkgs/stdenv/generic/make-derivation.nix:142:17, called from undefined position:
value is a string with context while a set was expected, at /home/builder/conduit/shell.nix:54:27
The command '/bin/sh -c nix-shell --show-trace --argstr type build' returned a non-zero code: 1
```

Reply:
I guess I can use `runDeps = with pkgs; [ unstable.poetry curl postgresql_11 ];` now. Yet getting the same error.

Reply:
It's not `unstable = fetchTarball { ... }`, see my comment above again

Reply:
Right missed that... Looks like it's working... Let's wait for the final result.